File: templates/problems/19_acm_io/01_fixed_format/typescript/fixedFormat.test.ts
import { describe, it, expect } from 'vitest';
import { solve } from './fixedFormat';

describe('FixedFormat', () => {
    it('basic sum', () => {
        expect(solve("5\n1 2 3 4 5\n")).toBe("15\n");
    });

    it('negative numbers', () => {
        expect(solve("3\n-1 0 1\n")).toBe("0\n");
    });

    it('single element', () => {
        expect(solve("1\n42\n")).toBe("42\n");
    });

    it('large values', () => {
        expect(solve("2\n1000000000 -1000000000\n")).toBe("0\n");
    });
});

File: templates/problems/19_acm_io/01_fixed_format/go/fixed_format_test.go
package fixed_format

import (
	"bufio"
	"bytes"
	"strings"
	"testing"
)

func TestSolve(t *testing.T) {
	tests := []struct {
		name  string
		input string
		want  string
	}{
		{"basic sum", "5\n1 2 3 4 5\n", "15\n"},
		{"negative numbers", "3\n-1 0 1\n", "0\n"},
		{"single element", "1\n42\n", "42\n"},
		{"large values", "2\n1000000000 -1000000000\n", "0\n"},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			reader := bufio.NewReader(strings.NewReader(tt.input))
			var buf bytes.Buffer
			writer := bufio.NewWriter(&buf)
			solve(reader, writer)
			writer.Flush()
			got := buf.String()
			if got != tt.want {
				t.Errorf("got %q, want %q", got, tt.want)
			}
		})
	}
}